### ISSUE TITLE
Rank English results by definition containment

### ISSUE THREAD

**Issue by Madoshakalaka** [UAlbertaALTLab/cree-intelligent-dictionary#196]

**Comment by eddieantonio** [comment]
Could you please explain what you mean by this issue? Perhaps given an example?

**Comment by Madoshakalaka** [comment]
When the user searches with an English word. The searched results should be reverse ranked by the proportion the user query is to the whole definition.

**Comment by eddieantonio** [comment]
Good point! I've added this to the [task board](https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/projects/6#card-30702417)!

**Comment by aarppe** [comment]
Unfortunately this won't work, since the prototypicality of a lexicon entry does not correlate with the shortness of the English gloss/definition. In fact, more prototypical words may have longer explanations, and this scheme would rank them lower, e.g. _wâpamêw_ or _wâpahtam_ (or _wâpiw_) for seeing, cf.

```
~/Google\ Drive/CreeFST/Dictionaries/Wolvengrey_eng2crk.csv | gawk -F"\t" '{ print $1"\t"$4; }' | egrep '\<see[s]?\>' | gawk -F"\t" '{ print length($2)"\t"$1"\t"$2; }' | sort -k1,1n | less
9       cîst    see here!
9       âsê-    see: asê-
15      mahci   come, let's see
17      mahti êsa       please; let's see
19      isinawêw        s/he sees s.o. thus
19      pimâpamêw       s/he sees s.o. pass
19      wâpihêw s/he makes s.o. see
20      wâpahtowak      they see one another
20      wâpamitowak     they see one another
21      pêtisâpahtam    s/he sees s.t. coming
21      pêtisâpamêw     s/he sees s.o. coming
21      pêtâpahtam      s/he sees s.t. coming
21      pêtâpamêw       s/he sees s.o. coming
21      wâpamisow       s/he sees him/herself
22      pakaskinam      s/he sees s.t. clearly
24      pê-wâpahtam     s/he comes and sees s.t.
24      pê-wâpamêw      s/he comes and sees s.o.
24      pîtotâpisin     s/he sees strange sights
25      atimâpamêw      s/he sees s.o. going away
25      iskwâpiw        s/he sees such a distance
25      ocâpahtam       s/he is alive to see s.t.
25      ocâpamêw        s/he is alive to see s.o.
28      kîsikânam       s/he sees (s.t. as) daylight
30      ocâpahtowin     being alive to see one another
30      wâpênam s/he sees s.t. as almost white
33      ocâpahtowak     they are alive to see one another
33      wâpahtamawêw    s/he sees (it/him) on or for s.o.
33      wâpênêw s/he sees s.o. as light in colour
34      macâpisin       s/he disapproves of what s/he sees
35      mâcikôtitân     look, let me show you; you will see
35      wâpahtam        s/he sees s.t., s/he witnesses s.t.
35      wâpamêw s/he sees s.o., s/he witnesses s.o.
```
The best solution for this would be a dedicated English-to-Cree dictionary, but that would require an amount of manual work that cannot be undertaken (by us) at this point in any reasonable time.

Therefore, the currently defined solution to this is observed corpus frequency of the Cree lexical entries (whose English definitions have a match for the English search string) as a proxy for prototypicality. A number of lexical entries do not have any occurrences in the A-W corpus, so we have to come up with an alternate scheme for them - one possibility is to use morpheme count, or a weighting based on frequencies of individual morphemes and their co-occurrences.

This is tied with #163 and has been explicitly mentioned (as "_corpus-based Cree lemma frequency for English searches_") in: https://docs.google.com/document/d/1tg9WoiGNy6J9Zaf2lwIXr0sWB0pFabn7hsOG50KxjC4/edit

**Comment by aarppe** [comment]
This has now been superseded by the ranking implemented in #197.

**Comment by aarppe** [comment]
Note - if we implement search with multiple words, then we could select Cree lexical entries based on an intersection of the matching of the individual English search words with the glosses for the lexical entries. Even so, we might want to more finely rank the matching Cree words according to some heuristic as the one implemented in #197.

**Comment by aarppe** [comment]
Duplicate of #197.
